rpcd/playbooks/ironic-pre-config.yml

```diff
@@ -0,0 +1,125 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+- name: Pre install config for ironic
+  hosts: ironic_all
+  tasks:
+    - name: Define networks for ironic to use
+      config_template:
+        src: "/etc/openstack_deploy/openstack_user_config.yml"
+        dest: "/etc/openstack_deploy/openstack_user_config.yml"
+        owner: "root"
+        group: "root"
+        mode: "0644"
+        config_overrides: "{{ cidr_networks_ironic }}"
+        config_type: "yaml"
+
+    - name: Add IPs to used_ips list
+  
+    - name: Setup global overrides
+      config_template:
+        src: "/etc/openstack_deploy/openstack_user_config.yml"
+        dest: "/etc/openstack_deploy/openstack_user_config.yml"
+        owner: "root"
+        group: "root"
+        mode: "0644"
+        config_overrides: "{{ ironic_networks }}"
+        config_type: "yaml"
+
+    - name: Define group hosts
+      config_template:
+        src: "/etc/openstack_deploy/openstack_user_config.yml"
+        dest: "/etc/openstack_deploy/openstack_user_config.yml"
+        owner: "root"
+        group: "root"
+        mode: "0644"
+        config_overrides: "{{ group_hosts_ironic }}"
+        config_type: "yaml"
+
+    - name: Setup endpoints for swift/ironic health checks
+       config_template:
+        src: "/etc/openstack_deploy/user_osa_variables_overrides.yml"
+        dest: "/etc/openstack_deploy/user_osa_variables_overrides.yml"
+        owner: "root"
+        group: "root"
+        mode: "0644"
+        config_overrides: "{{ ironic_endpoint_config }}"
+        config_type: "yaml"
+      
+    - name: Configure neutron DHCP to be non-authorative
+      config_template:
+        src: "/etc/openstack_deploy/user_osa_variables_overrides.yml"
+        dest: "/etc/openstack_deploy/user_osa_variables_overrides.yml"
+        owner: "root"
+        group: "root"
+        mode: "0644"
+        config_overrides: "{{ ironic_neutron_dhcp_config }}"
+        config_type: "yaml"
+
+    - name: Setup neutron network
+      shell: "neutron net-create --shared --provider:physical_network tftp --provider:network_type flat tftp"
+
+    - name: Setup neutron subnet
+      shell: "neutron subnet-create --name ironic-tftp --allocation-pool start=<NON_ALLOCATED_TFTP_RANGE_START>,end=<NON_ALLOCATED_TFTP_RANGE_END> --dns-nameserver=4.4.4.4 tftp <TFTP_NETWORK>/<TFTP_NETMASK>"
+
+  vars:
+    cidr_networks_ironic: 
+      cidr_networks:
+           tftp: {{ tftp_network }}{{ tftp_netmask }}
+           ironic-ipmi: {{ ironic-ipmi_network }}{{ ironic-ipmi_netmask }}
```

Comment:
`{{ ironic-ipmi_network }}{{ ironic-ipmi_netmask }}` and `{{ tftp_network }}{{ tftp_netmask }}` need to be surrounded by quotes.